### PR TITLE
Add ENVs using NewTestDockerCluster

### PR DESCRIPTION
**Pull request by gkoutsou**
### Description
Currently NewTestDockerCluster had no means for setting any environment variables. This makes it tricky to create test for functionality that require thems, like having to set AWS environment variables.

DockerClusterOptions now exposes an option to pass extra enviroment variables to the containers, which are appended to the existing ones.

Related to: https://github.com/hashicorp/vault/issues/26852

### TODO only if you're a HashiCorp employee
- [ ] **Changelog:** Make sure there is one, or the `pr/no-changelog` label is
  applied. Brand new features have a differently
  formatted changelog than other changelogs, so pay attention to that. If this
  PR is the CE portion of an ENT PR, put the changelog here, _not_ in your ENT
  PR. Feel free to refer to the [instructions](https://github.com/hashicorp/vault/blob/main/CONTRIBUTING.md#changelog-entries) on changelog formatting if necessary.
- [ ] **Labels:** If this PR is the CE portion of an ENT change, and that ENT change is
  getting backported to N-2, use the new style `backport/ent/x.x.x+ent` labels
  instead of the old style `backport/x.x.x` labels.
- [ ] **Labels:** If this PR is a CE only change, it can only be backported to N, so use
  the normal `backport/x.x.x` label (there should be only 1).
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description.

### CODE REVIEW

**Review by akshya96 — COMMENTED** [review]
Thanks for the contribution @gkoutsou . Could you please add a test to verify adding ENVs is working correctly?

**Comment by gkoutsou** [comment]
@akshya96 is that what you have in mind?

File: changelog/27457.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+skd/helper: Add ENVs using NewTestDockerCluster
```
Comment (akshya96) [inline changelog/27457.txt:2]:
```suggestion
sdk/helper: Add ENVs using NewTestDockerCluster
```

File: sdk/helper/testcluster/docker/environment_test.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package docker
+
+import (
+	"testing"
+)
+
+func TestSettingEnvsToContainer(t *testing.T) {
+	expectedEnv := "TEST_ENV=value1"
+	expectedEnv2 := "TEST_ENV=value2"
```
Comment (akshya96) [inline sdk/helper/testcluster/docker/environment_test.go:9]:
```suggestion
	expectedEnv2 := "TEST_ENV2=value2"
```

File: sdk/helper/testcluster/docker/environment_test.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		ImageRepo: "hashicorp/vault",
+		ImageTag:  "latest",
+		Envs:      []string{expectedEnv, expectedEnv2},
+	}
+	cluster := NewTestDockerCluster(t, opts)
+	defer cluster.Cleanup()
+
+	envs := cluster.GetActiveClusterNode().Container.Config.Env
+
+	if !findEnv(envs, expectedEnv) {
```
Comment (akshya96) [inline sdk/helper/testcluster/docker/environment_test.go:20]: In this test, won't value2 replace value1 for TEST_ENV? How can we obtain different values for the same ENV variable?

File: changelog/27457.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+sdk/helper: Add ENVs using NewTestDockerCluster
```
Comment (akshya96) [inline changelog/27457.txt:2]:
```suggestion
sdk/helper: Allow setting environment variables when using NewTestDockerCluster
```

**Review by akshya96 — APPROVED** [review]
I have a suggestion for changelog; everything else looks good to me.

**Comment by gkoutsou** [comment]
I noticed that the job that checked for missing copyright info failed so I just added it

**Pull request merged**